functions.php

```diff
@@ -32,6 +32,8 @@
 
 $google_maps_endpoint = "https://maps.googleapis.com/maps/api/geocode/json?key=" . trim($google_maps_api_key);
 $timezone_lookup_endpoint = "https://maps.googleapis.com/maps/api/timezone/json?key" . trim($google_maps_api_key);
+$osm_maps_endpoint = "https://nominatim.openstreetmap.org/search?format=json&limit=1&addressdetails=1";  
+$osm_timezone_lookup_endpoint = "https://nominatim.openstreetmap.org/search?format=json&limit=1&addressdetails=1";  
```

Comment:
could you escapsulate the selection of the endpoint into its own function?  this makes it re-usable without having to litter application logic with the endpoint selection.

Reply:
let me rephrase this, write a generic function that calls out either maps or timezone endpoints and maps to consistent response values.  decouple from application logic.